cms/private_media/storages.py

```diff
@@ -0,0 +1,41 @@
+import logging
+from typing import TYPE_CHECKING
+
+from botocore.exceptions import ClientError
+from storages.backends.s3 import S3Storage
+
+if TYPE_CHECKING:
+    from django.db.models.fields.files import FieldFile
+
+
+logger = logging.getLogger(__name__)
+
+
+class PrivacySettingS3Storage(S3Storage):
```

Comment:
@MebinAbraham NOTE: The ACL setting is now the responsibility of the 'Storage' class as outlined in requirements, and I've adjusted the language throughout to use 'set file permissions' instead of 'set file ACLs'. However, If we want to use other storage backends in future, they'll need to be subclassed and extended in a similar fashion.